### PR TITLE
adding recipe for kmc kmer counter

### PR DESCRIPTION
**Pull request by notestaff**
* [X] I have read the guidelines above.
* [X] This PR adds a new recipe.
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by notestaff** [comment]
@bioconda/core

File: recipes/kmc/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+build:
+  number: 2
```
Comment (bgruening) [inline recipes/kmc/meta.yaml]: Build number should start with 0, if this is a new package.

**Comment by bgruening** [comment]
Is it possible to compile this package and not use precompiled binaries?

File: recipes/kmc/build.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+set -efu -o pipefail
+
+export CPPFLAGS="-I${PREFIX}/include"
+export LDFLAGS="-L${PREFIX}/lib"
+
+echo src dir is ${SRC_DIR}
```
Comment (bgruening) [inline recipes/kmc/build.sh]: Can we remove the debug statements.

**Comment by bgruening** [comment]
@notestaff this looks really nice. Thanks for changing to compile the recipe!

**Pull request closed without merging**